flocker/control/_model.py

```diff
@@ -210,7 +216,10 @@ class Link(PRecord):
     """
     local_port = field(mandatory=True, type=int)
     remote_port = field(mandatory=True, type=int)
-    alias = field(mandatory=True)
+    alias = field(mandatory=True, factory=lambda s: s.lower(),
+                  invariant=lambda s: (
+                      (re.match(u"[a-zA-Z0-9]*$", s) is not None,
```

Comment:
Slight difference in semantics ... `u""` matches `u"[a-zA-Z0-9]*$"` but `u"".isalnum()` is `False`.  That's probably a change for the better, though.